dump/match_helper.py

```diff
@@ -0,0 +1,81 @@
+import os, sys, json
+from dump.match_infra import MatchRequest
+from dump.helper import create_template_dict
+
+# Port Helper Methods
+
+def fetch_port_oid(match_engine, port_name, ns):
+    """
+    Fetches thr relevant SAI_OBJECT_TYPE_PORT given port name
+    """
+    req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_HOSTIF", key_pattern="*", field="SAI_HOSTIF_ATTR_NAME",
+                       value=port_name, return_fields=["SAI_HOSTIF_ATTR_OBJ_ID"], ns=ns)
+    ret = match_engine.fetch(req)
+    asic_port_obj_id = ""
+    if not ret["error"] and len(ret["keys"]) != 0:
+        sai_hostif_obj_key = ret["keys"][-1]
+        if sai_hostif_obj_key in ret["return_values"] and "SAI_HOSTIF_ATTR_OBJ_ID" in ret["return_values"][sai_hostif_obj_key]:
+            asic_port_obj_id = ret["return_values"][sai_hostif_obj_key]["SAI_HOSTIF_ATTR_OBJ_ID"]
+    return req, asic_port_obj_id, ret
+
+# Vlan Helper Methods
+
+def fetch_vlan_oid(match_engine, vlan_name, ns):
+    # Convert 'Vlanxxx' to 'xxx'
+    if vlan_name[0:4] != "Vlan" or not vlan_name[4:].isnumeric():
+        vlan_num = -1
+    else:
+        vlan_num = int(vlan_name[4:])
+
+    # Find the table named "ASIC_STATE:SAI_OBJECT_TYPE_VLAN:*" in which SAI_VLAN_ATTR_VLAN_ID = vlan_num
+    req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_VLAN", key_pattern="*", field="SAI_VLAN_ATTR_VLAN_ID", 
+                    value=str(vlan_num), ns=ns)
+    ret = match_engine.fetch(req)
+    vlan_oid = ""
+    if ret["keys"]:
+        vlan_oid = ret["keys"][0].split(":", 2)[-1]
+    return req, vlan_oid, ret
+
+# LAG Helper Methods
+
+def get_lag_members_from_cfg(match_engine, lag_name, ns):
+    """
+    Get the members associated with a LAG from Config DB
+    """
+    lag_members = []
+    req = MatchRequest(db="CONFIG_DB", table="PORTCHANNEL_MEMBER", key_pattern=lag_name + "|*", ns=ns)
+    ret = match_engine.fetch(req)
+    for key in ret["keys"]:
+        lag_members.append(key.split("|")[-1])
+    return req, lag_members, ret
+
+def get_lag_and_member_obj(match_engine, port_asic_obj, ns):
+    """
+    Given the member port oid, fetch lag_member & lag oid's
+    """
+    req = MatchRequest(db="ASIC_DB", table="ASIC_STATE:SAI_OBJECT_TYPE_LAG_MEMBER", key_pattern="*", field="SAI_LAG_MEMBER_ATTR_PORT_ID",
+                        value=port_asic_obj, return_fields=["SAI_LAG_MEMBER_ATTR_LAG_ID"], ns=ns)
+    ret = match_engine.fetch(req)
+    lag_member_key = ""
+    lag_oid = ""
+    if not ret["error"] and ret["keys"]:
+        lag_member_key = ret["keys"][-1]
+        if lag_member_key in ret["return_values"] and "SAI_LAG_MEMBER_ATTR_LAG_ID" in ret["return_values"][lag_member_key]:
+            lag_oid = ret["return_values"][lag_member_key]["SAI_LAG_MEMBER_ATTR_LAG_ID"]
+    return lag_member_key, lag_oid
+
+def fetch_lag_oid(match_engine, lag_members, ns):
```

Comment:
Can we simplify getting lag_oid? Instead of user required to call get_lag_and_member_obj first and then pass the members to fetch_lag_oid can we make fetch_lag_oid internally call the members so users are abstracted?

Reply:
Handled